libs/@local/harpc/server/src/delegate.rs

```diff
@@ -0,0 +1,62 @@
+use alloc::sync::Arc;
+use core::{
+    fmt::Debug,
+    task::{Context, Poll},
+};
+
+use harpc_service::delegate::ServiceDelegate;
+use harpc_tower::{body::Body, request::Request, response::Response};
+use tower::Service;
+
+use crate::session::{Session, SessionStorage};
+
+#[derive_where::derive_where(Clone; D: Clone, C: Clone)]
+#[derive_where(Debug; D: Debug, S: Debug + 'static, C: Debug)]
+pub struct ServiceDelegateHandler<D, S, C> {
```

Comment:
Not a fan of `*Handler` structs, it's not clear what they are handling. Could you add documentation on what this does and why this is needed?

Reply:
not a fan of the name myself, but a `ServiceDelegateService` just sounded.... weird